libs/application/templates/parental-leave/src/fields/InReviewSteps/index.tsx

```diff
@@ -33,63 +56,124 @@ const statesMap: statesMap = {
+  const [screenState, setScreenState] = useState<'steps' | 'viewApplication'>(
+    'steps',
+  )
 
   return (
     <Box marginBottom={10}>
-      <Box>
-        <Text as="div">
-          {formatMessage(mm.reviewScreen.desc)}{' '}
+      <Box display="flex" justifyContent="spaceBetween">
+        <Text>
+          {(screenState === 'steps' && formatMessage(mm.reviewScreen.desc)) ||
+            formatMessage(mm.reviewScreen.descReview)}
+        </Text>
+        <Box>
           <Box display="inlineBlock" marginLeft={1} marginRight={2}>
             <Button
               colorScheme="default"
               iconType="filled"
-              // TODO: Add onClick in next PR
+              onClick={() =>
+                setScreenState(
+                  (screenState === 'steps' && 'viewApplication') || 'steps',
+                )
+              }
               size="small"
               type="button"
               variant="text"
             >
-              {formatMessage(mm.reviewScreen.buttonsView)}
+              {(screenState === 'steps' &&
+                formatMessage(mm.reviewScreen.buttonsView)) ||
+                formatMessage(mm.reviewScreen.buttonsViewProgress)}
             </Button>
           </Box>
           <Box display="inlineBlock">
-            <Button
-              colorScheme="default"
-              iconType="filled"
-              // TODO: Add onClick in next PR
-              size="small"
-              type="button"
-              variant="text"
-            >
-              {formatMessage(mm.reviewScreen.buttonsEdit)}
-            </Button>
+            <DialogPrompt
+              baseId="editApplicationDialog"
+              title={formatMessage(mm.reviewScreen.editApplicationModalTitle)}
+              description={formatMessage(
+                mm.reviewScreen.editApplicationModalDesc,
+              )}
+              ariaLabel={formatMessage(
+                mm.reviewScreen.editApplicationModalAria,
+              )}
+              disclosureElement={
+                <Button
+                  colorScheme="default"
+                  iconType="filled"
+                  size="small"
+                  type="button"
+                  variant="text"
+                  icon="pencil"
+                  loading={loadingSubmit}
+                  disabled={loadingSubmit}
+                >
+                  {formatMessage(mm.reviewScreen.buttonsEdit)}
+                </Button>
+              }
+              onConfirm={async () => {
+                const res = await submitApplication({
+                  variables: {
+                    input: {
+                      id: application.id,
+                      event: 'EDIT',
+                      answers: application.answers,
+                    },
+                  },
+                })
+
+                if (res?.data) {
+                  // Takes them back to the editable Review screen
+                  window.location.reload()
```

Comment:
I wonder if somehow, this could be using the `moveToScreen` util we have inside the ui-shell?

Reply:
Good point, curious how we could make it work:
When we submit the update to make the state machine go back to draft it will load a different form: From the `InReview` form to `ParentalLeaveForm`, so I'm not sure if we have access to those screens during this event.

Will confirm with Baering about the state machine change.

Reply:
I ended up going with a `refetch` approach, which should cause the re-render and avoid full page reload. Good idea to avoid full page reload.